xarray/coding/times.py

```diff
@@ -628,25 +628,37 @@ def _encode_datetime_with_cftime(dates, units: str, calendar: str) -> np.ndarray
     if cftime is None:
         raise ModuleNotFoundError("No module named 'cftime'")
 
+    dates = np.array(dates)
+
+    if dates.shape == ():
+        dates = dates.reshape(1)
```

Comment:
If you adapt the shape here, please revert this before returning result array. This will fix the backend roundtrip tests.